src/solver/simulation/simulation-runner.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include "include/antares/solver/simulation/simulation-runner.h"
+
+namespace Antares::Solver
+{
+
+SimulationRunner::SimulationRunner(Antares::Data::Study& study,
+                                   const Settings& settings,
+                                   Benchmarking::DurationCollector& durationCollector,
+                                   IResultWriter& resultWriter,
+                                   Simulation::ISimulationObserver& simulationObserver):
+    study_(study),
+    settings_(settings),
+    durationCollector_(durationCollector),
+    resultWriter_(resultWriter),
+    simulationObserver_(simulationObserver)
+{
+}
+
+Benchmarking::OptimizationInfo SimulationRunner::run()
+{
+    study_.computePThetaInfForThermalClusters();
+
+    if (study_.runtime.mode == Data::SimulationMode::Adequacy)
+    {
+        return runSimulation<Solver::Simulation::ISimulation<Solver::Simulation::Adequacy>>();
+    }
```

Comment:
Please add else if case for economy for clarity.
Alternatively we can define a map <simulationMode, simulationtype>

Reply:
Since runtime.mode is an enum it would be better to keep handling it with a switch case. And an except/error/log in the default case.
If we add a new mode it will default to economy in this case which may not be intended

Reply:
Done (**if** replaced with a **switch**)
If we add a new mode, we'll have to make changes in many places.
We'll take (I hope) this opportunity to use polymorphism instead of switching or use if everywhere.